nsv13/code/game/objects/effects/custom_landmarks.dm

```diff
@@ -0,0 +1,64 @@
+//Plutonium sludge
+
+#define PLUTONIUM_SLUDGE_RANGE 5
+#define PLUTONIUM_SLUDGE_RANGE_STRONG 10
+#define PLUTONIUM_SLUDGE_RANGE_WEAK 3
+
+#define PLUTONIUM_SLUDGE_CHANCE 35
+
+
+/obj/effect/landmark/nuclear_waste_spawner //Clean way of spawning nuclear gunk after a reactor core meltdown.
+	name = "nuclear waste spawner"
+	var/range = PLUTONIUM_SLUDGE_RANGE //tile radius to spawn goop
+	var/center_sludge = TRUE // Whether or not the center turf should spawn sludge or not.
+	var/static/list/avoid_objs = typecacheof(list( // List of objs that the waste does not spawn on
+		/obj/structure/stairs, // Sludge is hidden below stairs
+		/obj/structure/ladder, // Going down the ladder directly on sludge bad
+		/obj/effect/decal/nuclear_waste, // No stacked sludge
+		))
+
+/obj/effect/landmark/nuclear_waste_spawner/proc/fire()
+	playsound(loc, 'sound/effects/gib_step.ogg', 100)
+
+	if(center_sludge)
+		place_sludge(get_turf(src), TRUE)
+
+	for(var/turf/open/floor in orange(range, get_turf(src)))
+		place_sludge(floor, FALSE)
+
+	qdel(src)
+
+/// Tries to place plutonium sludge on 'floor'. Returns TRUE if the turf has been successfully processed, FALSE otherwise.
+/obj/effect/landmark/nuclear_waste_spawner/proc/place_sludge(turf/open/floor, epicenter = FALSE)
+	if(!floor)
+		return FALSE
+
+	if(epicenter)
+		for(var/obj/effect/decal/nuclear_waste/waste in floor) //Replace nuclear waste with the stronger version
+			qdel(waste)
+		new /obj/effect/decal/nuclear_waste/epicenter (floor)
+		return TRUE
+
+	if(!prob(PLUTONIUM_SLUDGE_CHANCE)) //Scatter the sludge, don't smear it everywhere
+		return TRUE
+
+	for(var/obj/O in floor)
+		if(O.density && O.anchored) // Dense and anchored objects block sludge (ex. machines, windows) while others (ex. canisters, cell chargers) do not.
```

Comment:
I think it would be better to just include grilles, girders, windows, the gateway and gravity generator in the avoid_objs list, because right now this stops sludge from spawning on way too many things, including but not limited to:
1. Directional windows
2. Doors & closed directional firedoors
3. Any machine or computer
4. Tables
5. Wrenched down crates and (closed) lockers
With this behaviour I would not be surprised if someone gets the brilliant idea to cover the entire ship in tables to avoid sludge spawning at all.